library/src/main/java/com/timehop/stickyheadersrecyclerview/HeaderPositionCalculator.java

```diff
@@ -0,0 +1,209 @@
+    MarginLayoutParams stickyHeaderLayout = (MarginLayoutParams) currentHeader.getLayoutParams();
+    if (orientation == LinearLayoutManager.VERTICAL) {
+      int topOfStickyHeader = getListTop(recyclerView) + stickyHeaderLayout.topMargin + stickyHeaderLayout.bottomMargin;
+      int shiftFromNextHeader = viewAfterNextHeader.getTop() - nextHeader.getHeight() - nextHeaderLayout.bottomMargin - nextHeaderLayout.topMargin - currentHeader.getHeight() - topOfStickyHeader;
+      if (shiftFromNextHeader < topOfStickyHeader) {
+        translation.top += shiftFromNextHeader;
+      }
+    } else {
+      int leftOfStickyHeader = getListLeft(recyclerView) + stickyHeaderLayout.leftMargin + stickyHeaderLayout.rightMargin;
+      int shiftFromNextHeader = viewAfterNextHeader.getLeft() - nextHeader.getWidth() - nextHeaderLayout.rightMargin - nextHeaderLayout.leftMargin - currentHeader.getWidth() - leftOfStickyHeader;
+      if (shiftFromNextHeader < leftOfStickyHeader) {
+        translation.left += shiftFromNextHeader;
+      }
+    }
+  }
+
+  /**
+   * Returns the first item currently in the RecyclerView that is not obscured by a header.
+   *
+   * @param parent Recyclerview containing all the list items
+   * @return first item that is fully beneath a header
+   */
+  private View getFirstViewUnobscuredByHeader(RecyclerView parent, View firstHeader) {
+    for (int i = 0; i < parent.getChildCount(); i++) {
+      View child = parent.getChildAt(i);
+      if (!itemIsObscuredByHeader(child, firstHeader, mOrientationProvider.getOrientation(parent))) {
+        return child;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Determines if an item is obscured by a header
+   *
+   * @param item        to determine if obscured by header
+   * @param header      that might be obscuring the item
+   * @param orientation of the {@link RecyclerView}
+   * @return true if the item view is obscured by the header view
+   */
+  private boolean itemIsObscuredByHeader(View item, View header, int orientation) {
+    RecyclerView.LayoutParams layoutParams = (RecyclerView.LayoutParams) item.getLayoutParams();
+    MarginLayoutParams headerLayout = (MarginLayoutParams) header.getLayoutParams();
+    if (orientation == LinearLayoutManager.VERTICAL) {
+      int itemTop = item.getTop() - layoutParams.topMargin;
+      int headerBottom = header.getBottom() + headerLayout.bottomMargin + headerLayout.topMargin;
+      if (itemTop > headerBottom) {
+        return false;
+      }
+    } else {
+      int itemLeft = item.getLeft() - layoutParams.leftMargin;
+      int headerRight = header.getRight() + headerLayout.rightMargin + headerLayout.leftMargin;
+      if (itemLeft > headerRight) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  private int getListTop(RecyclerView view) {
+    MarginLayoutParams layoutParams = (MarginLayoutParams) view.getLayoutParams();
+    int top = view.getTop() - layoutParams.topMargin;
```

Comment:
This (and the left calculation below) are what were breaking the headers in Timehop.  Is this part of what you did to support padding?  In my local branch I made `getListTop` and `getListLeft` methods return 0 to get things working again.